livekit-agents/livekit/agents/voice/room_io/_output.py

```diff
@@ -114,6 +117,11 @@ async def _wait_for_playout(self) -> None:
         self._interrupted_event.clear()
         self.on_playback_finished(playback_position=pushed_duration, interrupted=interrupted)
 
+    def _on_reconnected(self) -> None:
+        if self._republish_task:
+            self._republish_task.cancel()
+        self._republish_task = asyncio.create_task(self._publish_track())
```

Comment:
I generally pass the old task as a parameter to _publish_track to make sure to await it until the end.